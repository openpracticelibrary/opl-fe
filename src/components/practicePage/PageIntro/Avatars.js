import React from 'react';
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemText from "@material-ui/core/ListItemText";
import Avatar from "@material-ui/core/Avatar";
import Box from '@material-ui/core/Box';
import {makeStyles} from "@material-ui/core/styles/index";

const useStyles = makeStyles((theme) => ({
    small: {
        width: theme.spacing(3),
        height: theme.spacing(3),
    },
}));

export default function Avatars(props) {

    return (
        <React.Fragment>
            <Box>
                <ListItem>
                    <ListItemAvatar>
                        <Avatar
                            alt={ props.authorName }
                            src="/static/images/avatar/1.jpg"
                        />
                    </ListItemAvatar>
                    <ListItemText secondary={ props.authorName } />
                </ListItem>
            </Box>
        </React.Fragment>
    );
}
