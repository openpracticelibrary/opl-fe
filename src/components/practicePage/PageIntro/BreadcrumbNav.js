import React from "react";
import { makeStyles } from "@material-ui/core/styles/index";
import Breadcrumbs from "@material-ui/core/Breadcrumbs";
import Link from "@material-ui/core/Link";
import Box from "@material-ui/core/Box";

const useStyles = makeStyles((theme) => ({
  space: {
    paddingLeft: theme.spacing(1),
    marginBottom: theme.spacing(3),
  },
}));

export default function BreadcrumbNav() {
  const classes = useStyles();

  return (
    <>
      <Box className={classes.space}>
        <Breadcrumbs aria-label="breadcrumb">
          <Link variant="overline" href="/">
            Discovery
          </Link>
          <Link variant="overline" href="/">
            Validate
          </Link>
        </Breadcrumbs>
      </Box>
    </>
  );
}
