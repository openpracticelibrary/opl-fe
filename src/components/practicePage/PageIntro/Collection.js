import React from "react";
import { Stack, IconButton, Box, Text } from "@chakra-ui/core";
import { useMutation } from "@apollo/react-hooks";
import { LIKE_PRACTICE } from "../../../graphql/";

export default function Collection(props) {
  const [likePractice] = useMutation(LIKE_PRACTICE);

  const handleLike = () => {
    const originalLikes = props.upvotes;
    const newUpvotes = originalLikes + 1;
    likePractice({
      variables: {practiceId: props.practiceId, upvotes: newUpvotes},
    });
  };

  return (
    <Stack isInline pt={4} pb={8}>
      <Text
        fontFamily="heading"
        fontSize="sm"
        fontWeight="500"
        textTransform="uppercase"
      >
          Collection
      </Text>
      <Box verticalAlign="middle">
        <Text as="b" fontFamily="heading" data-testid="heartIcon">
          <IconButton
            isRound
            onClick={handleLike}
            height="auto"
            variant="ghost"
            aria-label="Like Practice"
            icon="like"
            color="grey.500"
            data-testid={"heartIcon"}
          />
          {props.upvotes}
        </Text>
        <Text as="b" fontFamily="heading">
          <IconButton
            isRound
            variant="ghost"
            height="auto"
            size="lg"
            aria-label="Image Count"
            color="grey.500"
            icon="camera"
          />
          {props.imgCount}
        </Text>
        <Text as="b" fontFamily="heading">
          <IconButton
            isRound
            variant="ghost"
            height="auto"
            aria-label="Questions Count"
            color="grey.500"
            icon="questions"
          />
          {props.questions}
        </Text>
      </Box>
    </Stack>
  );
}
