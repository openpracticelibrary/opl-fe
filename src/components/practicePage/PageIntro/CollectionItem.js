import React from 'react';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import ImageIcon from '@material-ui/icons/Image';
import {makeStyles} from "@material-ui/core/styles/index";

const useStyles = makeStyles((theme) => ({
    wrapIcon: {
        verticalAlign: 'middle',
        display: 'inline-flex',
        marginLeft: theme.spacing(2),
        color: theme.palette.common.brownish_grey,
    }
}));

export default function CollectionItem() {
    const classes = useStyles();

    return (
        <React.Fragment>
            <Box className={classes.wrapIcon}>
                <ImageIcon/>
                <Typography variant='body1'>12</Typography>
            </Box>
        </React.Fragment>
    );
}