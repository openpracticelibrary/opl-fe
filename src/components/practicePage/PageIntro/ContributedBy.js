import React from 'react';
import Typography from '@material-ui/core/Typography';
import {makeStyles} from "@material-ui/core/styles/index";
import PhotoAndName from './PhotoAndName';
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles((theme) => ({
    root: {
        flexGrow: 1,
        padding: theme.spacing(1),
    },
    space: {
        marginLeft: theme.spacing(7),
        alignContent: 'stretch'
    }
}));

export default function ContributedBy() {
    const classes = useStyles();

    return (
        <React.Fragment>
            <Grid container direction="column"
                  justify="center"
                  alignItems="flex-start"
                  className={classes.root}
            >
                <Grid item>
                    <Typography variant="overline">
                        Contributed by
                    </Typography>
                </Grid>

                <Grid item>
                    <Grid
                        container
                        direction="row"
                        justify="flex-start"
                        alignItems="center"
                    >
                        <Grid item>
                            <PhotoAndName/>
                        </Grid>
                        <Grid item>
                            <PhotoAndName/>
                        </Grid>
                        <Grid item className={classes.space}>
                            <Typography variant="overline">
                                Published May 4, 2020 | Last edited June 8, 2020
                            </Typography>
                        </Grid>
                    </Grid>
                </Grid>
            </Grid>
        </React.Fragment>
    );
}