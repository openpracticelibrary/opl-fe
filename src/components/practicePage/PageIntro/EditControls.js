import React from "react";
import {makeStyles} from "@material-ui/core/styles";
import CloseIcon from '@material-ui/icons/Close';
import SaveOutlinedIcon from '@material-ui/icons/SaveOutlined';
import { EditIcon } from "../../../assets/icons";

import { Tooltip, IconButton, Box } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  editingButtonCollection: {
    display: "flex",
    flexDirection: "column",
    backgroundColor: theme.palette.common.true_white,
    borderRadius: "20px",
    boxShadow: "0 1px 4px 0 rgba(16,16,16,0.29)",
  },
}));

const StartEditingButton = (props) => (
  <Tooltip
    title="Edit Practice"
    placement="top"
  >
    <IconButton data-testid="editButton" onClick={props.handleEdit}>
      <EditIcon height="20" />
    </IconButton>
  </Tooltip>
);

const EditingButtons = (props) => {
  const classes = useStyles();

  return (
    <Box className={classes.editingButtonCollection}>
      <IconButton>
        <EditIcon fill="#1975ff" height="20" />
      </IconButton>
      <Tooltip
        title="Save Edits"
        placement="left"
      >
        <IconButton data-testid="saveEditButton" onClick={props.handleSaveEdits}>
          <SaveOutlinedIcon />
        </IconButton>
      </Tooltip>
      <Tooltip
        title="Cancel Edits"
        placement="left"
      >
        <IconButton data-testid="closeEditButton" onClick={props.handleEdit}>
          <CloseIcon />
        </IconButton>
      </Tooltip>
    </Box>
  );
};

export { StartEditingButton, EditingButtons }
