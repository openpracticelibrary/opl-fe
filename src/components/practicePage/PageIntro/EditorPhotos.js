import React from "react";
import { makeStyles } from "@material-ui/core/styles/index";
import { AvatarGroup } from "@material-ui/lab";
import { Avatar } from "@material-ui/core";
import { UserIcon } from "../../../assets/icons";

const useStyles = makeStyles((theme) => ({
  small: {
    width: theme.spacing(3),
    height: theme.spacing(3),
    backgroundColor: "transparent",
  },
}));

const EditorPhotos = props => {
  const classes = useStyles();

  return (
    <AvatarGroup max={3}>
      {props.editors.map(editor => (
        <Avatar key={editor.id} src={editor.Avatar} className={classes.small}>
          <UserIcon />
        </Avatar>
      ))}
    </AvatarGroup>
  );
}

export default EditorPhotos;

