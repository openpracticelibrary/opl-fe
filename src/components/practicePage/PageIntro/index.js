import React from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import Container from '@material-ui/core/Container';
import {makeStyles} from "@material-ui/core/styles/index";
import Box from '@material-ui/core/Box';
import TitleAndSubtitle from './TitleAndSubtitle';
import ContributedBy from './ContributedBy';
import Collection from './Collection';
import BreadcrumbNav from './BreadcrumbNav';

const useStyles = makeStyles((theme) => ({
  root: {
    marginLeft: theme.spacing(2),
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
  },
  row: {
    display: 'flex',
    padding: theme.spacing(1),
    margin: theme.spacing(1)
  },
}));

export default function PageIntro(props) {
  const classes = useStyles();

  return (
      <>
        <CssBaseline/>
        <Container className={classes.root}>
          <main className={classes.content}>
            <BreadcrumbNav/>
            <TitleAndSubtitle title={props.title} subtitle={props.subtitle}/>
            <Box className={classes.row}>
              <ContributedBy
                  authors={props.authors}
                  createdAt={props.createdAt}
                  updatedAt={props.updatedAt}
              />
            </Box>
            <Box className={classes.row}>
              <Collection
                  practiceId={props.practiceId}
                  upvotes={props.upvotes}
                  imgCount={props.imgCount}
                  questions={props.questions}
                  handleLike={props.handleLike}
                  practice={props.practice}
              />
            </Box>

          </main>
        </Container>
      </>
  );
}
