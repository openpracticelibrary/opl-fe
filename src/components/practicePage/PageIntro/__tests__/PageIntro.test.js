import React from "react";
import ReactDOM from 'react-dom';
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import {MockedProvider} from "@apollo/react-testing";
import PageIntro from '../index';
import {GET_PRACTICE_PAGE} from "../../../../graphql";

const practice = [{
  __typename: "Practice",
  id: "5ec55c99d36579001355faae",
  title: "A title",
  subtitle: "A subtitle",
  coverImage: "https://fake.url/an_image.jpeg",
  authors: {
    id: "123",
    firstName: "testFirstName",
    lastName: "testLastName",
  },
  tags: [
    {
      __typename: "Tags",
      tag: "discovery",
    },
    {
      __typename: "Tags",
      tag: "value",
    },
  ],
  upvotes: 42,
  updatedAt: "2020-05-27T20:24:29.223Z",
  slug: "event-storming",
  mediaGallery: [
    {
      __typename: "ComponentPracticeBodyMediaGallery",
      link: "https://a.picture",
    },
    {
      __typename: "ComponentPracticeBodyMediaGallery",
      link: "https://a.video.thumbnail",
    },
  ],
  ama: [
    {
      __typename: "Questions",
      question: "When do I do this?",
    },
  ],
}];

const apolloMocks = [
  {
    request: {
      query: GET_PRACTICE_PAGE,
    },
    result: {data: {practice}}
  },
];

afterEach(cleanup);


it("renders BodyIntro without crashing", () => {

  console.log(">>>", practice[0].authors);

  const { getByTestId } = render(
    <MockedProvider mocks={apolloMocks}>
      <PageIntro props={practice[0]}/>
    </MockedProvider>);
});

