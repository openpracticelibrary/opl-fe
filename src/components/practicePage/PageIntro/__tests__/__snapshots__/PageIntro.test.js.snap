// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-layout-187 MuiGrid-container MuiGrid-direction-xs-column"
>
  <div
    className="MuiGrid-root MuiGrid-item"
  >
    <div
      className="MuiBox-root MuiBox-root-297 makeStyles-root-292"
    >
      <div
        className="MuiBox-root MuiBox-root-298 makeStyles-space-293"
      >
        <h2
          className="MuiTypography-root MuiTypography-h2"
          data-testid="title"
        >
          This is a test title
        </h2>
        <div
          className="MuiBox-root MuiBox-root-329 makeStyles-tagContainer-294"
        />
      </div>
      <div
        className="MuiBox-root MuiBox-root-330 makeStyles-space-293"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          data-testid="subtitle"
        >
          A subtitle for your title
        </h6>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-row-188 MuiGrid-item"
  >
    <div
      className="MuiGrid-root makeStyles-root-331 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-justify-xs-center"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
        >
          <div
            className="MuiBox-root MuiBox-root-335"
          >
            <span
              className="MuiTypography-root MuiTypography-overline"
            >
              Contributed by
            </span>
            <div
              className="MuiGrid-root MuiGrid-item"
            >
              <div
                className="MuiBox-root MuiBox-root-336 makeStyles-editorBox-333"
              >
                <div
                  className="MuiGrid-root makeStyles-space-337 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      className="MuiAvatar-root MuiAvatar-circle makeStyles-small-338 MuiAvatar-colorDefault"
                    >
                      <svg
                        height="24"
                        viewBox="0 0 25 24"
                        width="25"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12.696 0C6.084 0 .704 5.383.704 12s5.38 12 11.992 12c6.612 0 11.992-5.383 11.992-12S19.308 0 12.696 0zm2.058 17.213c1.863-1.444 3.17-4.571 3.17-6.771 0-2.946-2.341-5.334-5.228-5.334-2.887 0-5.227 2.388-5.227 5.334 0 2.2 1.306 5.327 3.169 6.771-2.584.34-4.647 1.308-5.49 2.551-1.874-1.948-3.033-4.616-3.033-7.556 0-5.953 4.747-10.796 10.581-10.796s10.58 4.843 10.58 10.796c0 2.94-1.158 5.608-3.032 7.557-.843-1.244-2.905-2.211-5.49-2.552z"
                          fill="#A0A0A0"
                          fillRule="nonzero"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1"
                      data-testid="authorName"
                    >
                      Tim Beattie
                    </p>
                  </div>
                </div>
                <div
                  className="MuiGrid-root makeStyles-space-337 MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <div
                      className="MuiAvatar-root MuiAvatar-circle makeStyles-small-338 MuiAvatar-colorDefault"
                    >
                      <svg
                        height="24"
                        viewBox="0 0 25 24"
                        width="25"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12.696 0C6.084 0 .704 5.383.704 12s5.38 12 11.992 12c6.612 0 11.992-5.383 11.992-12S19.308 0 12.696 0zm2.058 17.213c1.863-1.444 3.17-4.571 3.17-6.771 0-2.946-2.341-5.334-5.228-5.334-2.887 0-5.227 2.388-5.227 5.334 0 2.2 1.306 5.327 3.169 6.771-2.584.34-4.647 1.308-5.49 2.551-1.874-1.948-3.033-4.616-3.033-7.556 0-5.953 4.747-10.796 10.581-10.796s10.58 4.843 10.58 10.796c0 2.94-1.158 5.608-3.032 7.557-.843-1.244-2.905-2.211-5.49-2.552z"
                          fill="#A0A0A0"
                          fillRule="nonzero"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item"
                  >
                    <p
                      className="MuiTypography-root MuiTypography-body1"
                      data-testid="authorName"
                    >
                      Matt Takane
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiBox-root MuiBox-root-346"
          />
          <div
            className="MuiGrid-root makeStyles-space-332 MuiGrid-item"
          >
            <span
              className="MuiTypography-root MuiTypography-overline"
              data-testid="dates"
            >
              Published 
              July 8, 2020
               | Last edited
               
              July 8, 2020
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root makeStyles-row-188 MuiGrid-item"
  >
    <div
      className="MuiGrid-root makeStyles-root-347 MuiGrid-container"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <span
          className="MuiTypography-root MuiTypography-overline"
        >
          Collection
        </span>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root makeStyles-button-348"
          data-testid="heartIcon"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiGrid-root makeStyles-layout-361 MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <svg
                  height="17px"
                  version="1.1"
                  viewBox="0 0 20 17"
                  width="20px"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <title>
                    DB31F47E-55D5-437E-95A3-7A5BAD5EF398
                  </title>
                  <g
                    fill="#596562"
                    fillRule="evenodd"
                    id="Practice-Page---Product"
                    stroke="none"
                    strokeWidth="1"
                  >
                    <g
                      fillRule="nonzero"
                      id="OPL_Practice-Pg--All-Content-New-Link"
                      stroke="#596562"
                      transform="translate(-1235.000000, -423.000000)"
                    >
                      <g
                        id="Practice-Nav---Wide"
                        transform="translate(0.000000, 389.000000)"
                      >
                        <g
                          id="noun_Love_1969794"
                          transform="translate(1236.000000, 35.000000)"
                        >
                          <path
                            d="M17.9963086,4.81542188 C17.908418,2.15792578 15.8348672,0.00028125 13.1690742,0.00028125 C11.3911172,0.00028125 9.83826562,0.958253906 9,2.38422656 C8.16176953,0.958253906 6.60895312,0.00028125 4.83089062,0.00028125 C2.16464062,0.00028125 0.0910546875,2.15792578 0.0037265625,4.81542188 C-0.196207031,10.841168 9,15.1872188 9,15.1872188 C9,15.1872188 18.1956797,10.841168 17.9963086,4.81542188 Z"
                            id="Path"
                          />
                        </g>
                      </g>
                    </g>
                  </g>
                </svg>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <p
                  className="MuiTypography-root makeStyles-space-362 MuiTypography-body1"
                >
                  42
                </p>
              </div>
            </div>
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiGrid-root makeStyles-layout-361 MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <svg
                  height="16"
                  viewBox="0 0 22 16"
                  width="22"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="#A0A0A0"
                  >
                    <path
                      d="M19.565 2.754H16.45L14.493.724C14.058.29 13.478 0 12.899 0H8.84c-.58 0-1.16.29-1.595.725L5.29 2.754H2.174C.942 2.754 0 3.696 0 4.928v8.84c0 1.232.942 2.174 2.174 2.174h17.391c1.232 0 2.174-.942 2.174-2.174v-8.84c0-1.16-.942-2.174-2.174-2.174zm-8.695 10.87c-2.682 0-4.856-2.175-4.856-4.856s2.174-4.855 4.856-4.855c2.68 0 4.855 2.174 4.855 4.855 0 2.681-2.174 4.855-4.855 4.855z"
                    />
                    <circle
                      cx="10.87"
                      cy="8.768"
                      r="3.406"
                    />
                  </g>
                </svg>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <p
                  className="MuiTypography-root makeStyles-space-362 MuiTypography-body1"
                >
                  3
                </p>
              </div>
            </div>
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <div
              className="MuiGrid-root makeStyles-layout-361 MuiGrid-container"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <svg
                  height="15px"
                  version="1.1"
                  viewBox="0 0 18 15"
                  width="18px"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <title>
                    6E24265F-DCD4-4060-AD24-1D9A528732F8
                  </title>
                  <g
                    fill="none"
                    fillRule="evenodd"
                    id="Practice-Page---Product"
                    stroke="none"
                    strokeWidth="1"
                  >
                    <g
                      fill="#596562"
                      fillRule="nonzero"
                      id="OPL_Practice-Pg--All-Content"
                      transform="translate(-438.000000, -310.000000)"
                    >
                      <g
                        id="Practice-Intro---Wide"
                      >
                        <g
                          id="collection"
                        >
                          <g
                            transform="translate(265.815406, 304.000000)"
                          >
                            <g
                              id="icons"
                              transform="translate(87.938931, 0.000000)"
                            >
                              <g
                                id="noun_Question_2965389"
                                transform="translate(94.434421, 13.500000) scale(-1, 1) translate(-94.434421, -13.500000) translate(86.939625, 6.000000)"
                              >
                                <path
                                  d="M11.5828654,0 C12.4863851,0 13.352899,0.359171047 13.9917839,0.99849961 C14.6306688,1.63782817 14.9895906,2.50494378 14.9895906,3.40909091 L14.9895906,8.86363636 C14.9895906,10.7464253 13.4643478,12.2727273 11.5828654,12.2727273 L8.38054381,12.2727273 L4.46280992,14.8840909 C4.35195049,14.9586494 4.22163656,14.9989563 4.08807015,15 C3.97623357,14.9994408 3.86625402,14.9713414 3.76783799,14.9181818 C3.54616431,14.8000648 3.40739525,14.5694957 3.40672513,14.3181818 L3.40672513,12.2727273 C2.50320545,12.2727273 1.63669158,11.9135562 0.997806688,11.2742277 C0.358921795,10.6348991 -9.08863547e-13,9.76778349 -9.08863547e-13,8.86363636 L-9.08863547e-13,3.40909091 C-9.08863547e-13,1.52630199 1.52524279,0 3.40672513,0 L11.5828654,0 Z M7.40944938,8.57111111 C6.85427936,8.57111111 6.41014335,9.01555556 6.41014335,9.57111111 C6.41014335,10.1266667 6.85427936,10.5711111 7.40944938,10.5711111 C7.96461941,10.5711111 8.40875542,10.1266667 8.40875542,9.57111111 C8.40875542,9.01555556 7.96461941,8.57111111 7.40944938,8.57111111 Z M7.3720379,1.57111111 C6.92388781,1.57111111 6.53005895,1.64533792 6.1633907,1.78142039 C5.81030275,1.91750286 5.49795572,2.10306987 5.23992991,2.35049255 C4.98190411,2.59791523 4.77819952,2.88245132 4.64239647,3.21647194 C4.49301311,3.55049255 4.42511158,3.92162658 4.41153127,4.31750286 L4.41153127,4.31750286 L4.41924489,4.36035647 C4.4392351,4.4149874 4.49572917,4.45853379 4.56091463,4.47832761 C4.56091463,4.47832761 4.88684197,4.57729668 5.30783144,4.57729668 C5.72882092,4.57729668 6.05474825,4.47832761 6.05474825,4.47832761 C6.13623009,4.45358534 6.20413161,4.39172967 6.20413161,4.31750286 C6.20413161,4.31750286 6.21771192,3.93399771 6.27203314,3.76080183 C6.32635436,3.58760596 6.4078362,3.42678121 6.50289834,3.30306987 C6.61154078,3.17935853 6.73376353,3.06801833 6.8967272,2.99379152 C7.05969087,2.91956472 7.24981515,2.88245132 7.46710004,2.88245132 C7.79302737,2.88245132 8.05105318,2.96904926 8.22759715,3.129874 C8.40414112,3.29069874 8.49920326,3.55049255 8.49920326,3.88451317 C8.49920326,4.08245132 8.45846235,4.24327606 8.37698051,4.37935853 C8.28191837,4.51544101 8.17327593,4.63915235 8.03747287,4.75049255 C7.90166982,4.86183276 7.75228645,4.97317297 7.58932279,5.08451317 C7.42635912,5.19585338 7.27697576,5.33193585 7.1275924,5.48038946 C6.97820903,5.62884307 6.85598628,5.81441008 6.74734384,6.02471936 C6.63870139,6.24739977 6.57079987,6.50719359 6.54363925,6.82884307 L6.54363925,6.82884307 L6.54363925,7.32368843 C6.54363925,7.39791523 6.61154078,7.44739977 6.67944231,7.47214204 C6.67944231,7.47214204 6.99178934,7.57111111 7.3720379,7.57111111 C7.76586676,7.57111111 8.06463348,7.47214204 8.06463348,7.47214204 C8.14611532,7.44739977 8.20043654,7.39791523 8.20043654,7.32368843 L8.20043654,7.32368843 L8.20043654,6.90306987 C8.22759715,6.68038946 8.30907898,6.49482245 8.43130174,6.34636884 C8.55352449,6.19791523 8.70290785,6.06183276 8.86587151,5.95049255 C9.02883518,5.83915235 9.20537916,5.71544101 9.38192313,5.6041008 C9.57204741,5.48038946 9.73501107,5.34430699 9.88439444,5.18348225 C10.0337778,5.01028637 10.1695809,4.82471936 10.264643,4.58966781 C10.3732854,4.3669874 10.4276067,4.07008018 10.400446,3.69894616 C10.400446,3.48863688 10.3461248,3.25358534 10.2510627,3.00616266 C10.1560005,2.75873998 9.97945658,2.52368843 9.74859138,2.31337915 C9.50414588,2.10306987 9.19179885,1.929874 8.81155029,1.78142039 C8.43130174,1.64533792 7.94241073,1.57111111 7.3720379,1.57111111 Z"
                                  id="Shape"
                                  transform="translate(7.494795, 7.500000) scale(-1, 1) translate(-7.494795, -7.500000) "
                                />
                              </g>
                            </g>
                          </g>
                        </g>
                      </g>
                    </g>
                  </g>
                </svg>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <p
                  className="MuiTypography-root makeStyles-space-362 MuiTypography-body1"
                >
                  5
                </p>
              </div>
            </div>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
